soaks/tests/http_pipelines_blackhole/vector.toml

```diff
@@ -10,7 +10,7 @@ type = "internal_metrics"
 [sources.logs]
 type = "http"
 address = "0.0.0.0:8282"
-encoding = "text"
+encoding = "ndjson"
```

Comment:
One thing that's not entirely clear yet to me (I should find some time to run this soak locally and inspect the output), is what this change means for the data coming in.

That is, `ndjson` supports multiple JSON types separated by newlines. Are we taking a multi-line payload through the HTTP source, and splitting it up into multiple events here?

I'm assuming _no_, since I can't see how that would've worked with the VRL program(s) used below. So, I assume that we're just getting one JSON object per HTTP request, correct?

Reply:
I believe your reading is correct. Previously we were taking newline-delimited lines, but just reading them as raw text, one per event. Now it's the same, one line per event, they are just parsed as JSON.

Reply:
> One thing that's not entirely clear yet to me (I should find some time to run this soak locally and inspect the output), is what this change means for the data coming in.

I _think_ nothing, but confirmation is welcome. 

> That is, ndjson supports multiple JSON types separated by newlines. Are we taking a multi-line payload through the HTTP source, and splitting it up into multiple events here?
>
> I'm assuming no, since I can't see how that would've worked with the VRL program(s) used below. So, I assume that we're just getting one JSON object per HTTP request, correct?

Right, if the semantics are different here that'd be good to know. The incoming payload from lading is a newline delimited json blob, one json object per line. Previously, as I understand it, the text encoding split the lines and we then parse each string that happened to be a json object in the next step.